### DISCUSSION TITLE
Oh My Zsh plugins

### DISCUSSION THREAD

**Discussion by tomclark** [marlonrichert/zsh-launchpad#6] [Q&A] [answered]
Hi there,

Thanks for this really great template. :-)

I'm afraid I'm struggling to figure out how to use plugins from Oh My Zsh (specifically the [warp directory](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/wd) one). I figured I should add it to the array of plugins, but with the plugin being in a subdirectory on Github, that causes issues.

Could you point me in the right direction, please? It'd be good to understand how to source other things from OMZ too (lib and themes, for example). Other than that, I'm using the template as-is, so I've not modified anything funky.

Thanks again.

**Comment by tomclark** [comment, marked as answer]
The instant I posted that, I tried something else and got it working! Doh!

I was trying to do something like this, but because it's in a subdirectory on Github, you can't do it:

```
local -a plugins=(
    marlonrichert/zsh-autocomplete      # Real-time type-ahead completion
    marlonrichert/zsh-edit              # Better keyboard shortcuts
    marlonrichert/zsh-hist              # Edit history from the command line.
    marlonrichert/zcolors               # Colors for completions and Git
    zsh-users/zsh-autosuggestions       # Inline suggestions
    zsh-users/zsh-syntax-highlighting   # Command-line syntax highlighting
    ohmyzsh/ohmyzsh               # << You can't do this and then reference it later
)
```

Instead of trying to add OMZ to the list of plugins so it behaved like the others, I just sourced it separately in the plugins file at the bottom:

```
znap source ohmyzsh plugins/wd
```
And that works.

Apologies. Next time I'll spend an extra five minutes before asking a question.

Thanks again for the great template.

Edit: is it necessary to ```znap clone ohmyzsh/ohmyzsh```? It works as-is, so I guess not...but the others are cloned from the array.

**Reply by marlonrichert** [reply]
`znap source` automatically calls `znap clone`. So, no, it’s not strictly necessary. It just reduces your initial startup time to clone everything in parallel. However, once everything is cloned, that’s no longer an issue.

**Reply by marlonrichert** [reply]
@tomclark PS: I maintain all of my software for Zsh in my free time. If you enjoy using it, please consider giving back by [sponsoring me](https://github.com/sponsors/marlonrichert). 🙂
